service/resource/configmap/update.go

```diff
@@ -2,12 +2,64 @@ package configmap
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
 )
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
```

Comment:
Are the first and third return values never set to anything but `nil`? What is the reason for having them or am I missing something?

Reply:
Because this is the resource interface being implemented. Other operators need to implement resources that compute these values, for reasons. My thinking towards the create state here is that we can probably remove it. I need to think about this a bit more though and I want to get more experiences with as many as different operators and resources as possible to validate this decision. The delete state though is something we have to keep, because I see it being useful in several cases.

Reply:
Ack, wasn't aware of that :+1:

Reply:
There's also some consideration for this specific controller. The configuration for prometheus is supplied as part of the chart, so we don't need/want to create it, and we definitely don't want to delete it, so create/delete should always be no-ops.